lib/gnat/jetstream/api/kv.ex

```diff
@@ -248,10 +251,34 @@ defmodule Gnat.Jetstream.API.KV do
     Gnat.Jetstream.API.KV.Watcher.stop(pid)
   end
 
+  @doc ~S"""
+  Returns true if operation is `{"kv-operation", "DEL"}` or `{"kv-operation", "PURGE"}`
+
+  ## Parameters
+  - `headers` - a list of headers to test
+
+  ## Examples
+
+      iex> is_delete_operation?([{"kv-operation", "DEL"}])
+      true
+      iex> is_delete_operation?([{"kv-operation", "PURGE"}])
+      true
+      iex> is_delete_operation?([{"kv-operation", "ADD"}])
+      false
+  """
+  @spec is_delete_operation?(headers :: Gnat.headers()) :: boolean()
+  def is_delete_operation?(headers) do
+    headers
+    |> Enum.filter(fn {k, v} ->
+      k == @operation_header and (v == @operation_del or v == @operation_purge)
+    end)
+    |> length() > 0
```

Comment:
```suggestion
    Enum.any?(headers, fn {k, v} ->
      k == @operation_header and (v == @operation_del or v == @operation_purge)
    end)
```

In practice this probably isn't a huge issue as written since the size of headers is probably small, but this is equivalent. At worst it makes a full pass once, at best it short-circuits early